lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -110,15 +112,61 @@ def parse_namespace_metadata(namespace_object)
     def get_pod_metadata(namespace_name, pod_name)
       begin
         metadata = @client.get_pod(pod_name, namespace_name)
-        return if !metadata
-        return parse_pod_metadata(metadata)
-      rescue KubeException
+        unless metadata
+          @stats.bump(:pod_cache_api_nil_not_found)
+        else
+          begin
+            metadata = parse_pod_metadata(metadata)
+            @stats.bump(:pod_cache_api_updates)
+            return metadata
+          rescue Exception=>e
+            log.debug(e)
+            @stats.bump(:pod_cache_api_nil_bad_resp_payload)
+            nil
+          end
+        end
+      rescue KubeException=>e
+        @stats.bump(:pod_cache_api_nil_error)
+        log.debug "Exception encountered fetching pod metadata from Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}: #{e.message}"
+        nil
+      end
+    end
+
+    def dump_stats
+      @curr_time = Time.now
+      return if (@curr_time - @prev_time).to_i < @stats_interval
+      @prev_time = @curr_time
+      log.info(@stats)
```

Comment:
Can we also dump info about the caches themselves?  How many entries currently maybe?

Reply:
Not sure what the ask is here.  By dumping the stats you will get a list of all the entries and the associated counts.  What additional value is provided by the raw number you are requesting?

Reply:
I was thinking we should emit the size of each cache at the least, to know if we are actually evicting on new entries or only on ttl.